src/snowflake/snowpark/_internal/analyzer/analyzer.py

```diff
@@ -605,7 +605,7 @@ def table_function_expression_extractor(
             sql = named_arguments_function(
                 expr.func_name,
                 {
-                    key: self.analyze(
+                    key: self.to_sql_try_avoid_cast(
```

Comment:
to_sql_try_avoid_cast is for numeric types. Does it also work for boolean values?

Reply:
I added the support for boolean type in the to_sql_try_avoid_cast function